crates/sui-types/src/transaction.rs

```diff
@@ -2613,19 +2613,23 @@ impl TransactionDataAPI for TransactionDataV1 {
                 ));
             }
 
-            /* Initially, we validate that (current_epoch == min_epoch == max_epoch) for simplicity.
-            This is intentionally overly strict, we intend to relax these rules as needed. */
             match (min_epoch, max_epoch) {
                 (Some(min), Some(max)) => {
-                    if min != max {
+                    if config.enable_multi_epoch_transaction_expiration() {
+                        if min > max || *max > min.saturating_add(1) {
```

Comment:
nit: wouldn't it be clearer to write `max == min || max == min.saturating_add(1)`